sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -383,7 +383,7 @@ static int sof_audio_probe(struct platform_device *pdev)
 	INIT_LIST_HEAD(&ctx->hdmi_pcm_list);
 
 	sof_audio_card_pcm512x.dev = &pdev->dev;
-	mach = (&pdev->dev)->platform_data;
+	mach = pdev->dev.platform_data;
```

Comment:
it may be clumsy but that's what we use in all other machine drivers. Either we change them all or we don't

Reply:
done